airflow/providers/google/cloud/triggers/vertex_ai.py

```diff
@@ -16,19 +16,38 @@
 # under the License.
 from __future__ import annotations
 
-from typing import Any, AsyncIterator, Sequence
+from functools import cached_property
+from typing import TYPE_CHECKING, Any, AsyncIterator, Sequence
 
-from google.cloud.aiplatform_v1 import HyperparameterTuningJob, JobState
+from google.cloud.aiplatform_v1 import BatchPredictionJob, HyperparameterTuningJob, JobState, types
 
 from airflow.exceptions import AirflowException
+from airflow.providers.google.cloud.hooks.vertex_ai.batch_prediction_job import BatchPredictionJobAsyncHook
 from airflow.providers.google.cloud.hooks.vertex_ai.hyperparameter_tuning_job import (
     HyperparameterTuningJobAsyncHook,
 )
 from airflow.triggers.base import BaseTrigger, TriggerEvent
 
+if TYPE_CHECKING:
+    from proto import Message
 
-class CreateHyperparameterTuningJobTrigger(BaseTrigger):
-    """CreateHyperparameterTuningJobTrigger run on the trigger worker to perform create operation."""
+
+class BaseVertexAIJobTrigger(BaseTrigger):
+    """Base class for Vertex AI job triggers.
+
+    This trigger polls the Vertex AI job and checks its status.
+
+    In order to use it properly, you must:
+        - implement the following methods `_wait_job()`.
+        - override `trigger_class_path` attribute to provide correct path to your trigger class.
+        - override `job_type_verbose_name` attribute to provide meaningful message describing your job type.
+        - override `job_serializer_class` attribute to provide proto.Message class that will be used to
+        serialize your job with `to_dict()` class method.
+    """
+
+    trigger_class_path: str = ""
+    job_type_verbose_name: str = "Vertex AI Job"
+    job_serializer_class: Message = None
```

Comment:
If `job_serializer_class` default can be _None_, then the other class attributes can also be made equal to _None_?

Also, the typing hint for `job_serializer_class` should be `Message | None = None` or `Optional(Message) `.

Reply:
It can't be None, thus the typing is misleading. Thank you for noticing that. I fixed it by moving it to the `__init__()` method, where it is assigned automatically. So from now, it's free from human error, and developers don't even have to work with it.